pkg/cvo/sync.go

```diff
@@ -12,13 +12,20 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
 
 	"github.com/openshift/cluster-version-operator/lib"
 	"github.com/openshift/cluster-version-operator/lib/resourcebuilder"
 	cvv1 "github.com/openshift/cluster-version-operator/pkg/apis/config.openshift.io/v1"
 	"github.com/openshift/cluster-version-operator/pkg/cvo/internal"
 )
 
+const (
+	// RequeueOnErrorAnnotationKey is key for annotation on a manifests object that instructs CVO to requeue on specific errors.
+	// The value is comma separated list of causes that forces requeue.
+	RequeueOnErrorAnnotationKey = "v1.cluster-version-operator.operators.openshift.io/RequeueOnError"
```

Comment:
Suggest `requeue-on-error` which is the upstream convention we've mostly settled on for new things

Reply:
Also, define the set of supported causes here.

Reply:
:+1:

Reply:
done with https://github.com/openshift/cluster-version-operator/pull/57/commits/31e4d2957e789ff1251f3c14a5854d48202372b2